### DISCUSSION TITLE
Combination of media query and custom prefixed clases

### DISCUSSION THREAD

**Discussion by otakarprokes** [tailwindlabs/tailwindcss#11078] [Help] [answered]
Hello,
I try to create collapsible sidebar with tailwind. I found a lot of samples, but they always use only on 1 of these 2 options:

- collapse on media query
- collapse on user input - clicking the button

I did not find any solution, which combine both. Maybe the "DarkMode like" solution could be the path. 

![image](https://user-images.githubusercontent.com/47251544/233976694-525cf899-187e-47d3-84ff-954d30855009.png)

myClass:w-0 - this would have to have preference over md:w-60 every time myClass exists in parent element no matter what the window size is. Is this possible? Can I get some hints, how to set it up, if it is possible? Or is there another tailwind solution for that?

Thank you 
Ota

**Comment by wongjn** [comment, marked as answer]
You can indeed set up your own variants. See [the documentation on adding variants](https://tailwindcss.com/docs/plugins#adding-variants) for details.

**Reply by otakarprokes** [reply]
With your advice and information from link bellow I was able to achieve, what I want. Thank you
https://github.com/tailwindlabs/tailwindcss/pull/8299
